### ISSUE TITLE
view function returns only one element

### ISSUE THREAD

**Issue by MartinKavik** [David-OConnor/seed#88]
Hi,
can I return more elements from root `view`?

I think that `fn view(model: &Model) -> Vec<El<Msg>>` would correspond more with quote from Seed documentation: 

> Fragments (<>...</> syntax in React and Yew) are components that represent multiple elements without a parent. They're useful to avoid unecessary divs, which clutter teh DOM, and breaks things like tables and CSS-grid. 

Thanks!

**Comment by David-OConnor** [comment]
This is a good point... Would be easy to implement. Am willing to make this breaking change if it looks like the right approach.

**Comment by David-OConnor** [comment]
Do you see a practical advantage over the current method, or is it about elegance/consistency/something else? I see the consistency argument.

**Comment by MartinKavik** [comment]
I've encountered this problem in my projects:

1) Side-project currently written in Yew, I want to rewrite it to Seed (cca 6k LoC, website editor, probably will be open-source)
2) I've started to develop webpack-based alternative to seed-quickstart with tailwindcss and examples of tests, typescript integration and easy deployment into netlify.

I'm using array of elements in both project, so it's more practical than elegance problem for me.

**Comment by MartinKavik** [comment]
Inspiration:

I did a little research and there is an interesting approach to resolving `view` function in the latest [Elm.Browser](https://package.elm-lang.org/packages/elm/browser/latest/Browser#application). It has type `view : model -> Document msg` where `Document` is:
```elm
type alias Document msg =
    { title : String
    , body : List (Html msg)
    }
```
So it returns list of elements and even a new page title.

I think that with similar approach we can provide more flexibility and add more functionality (e.g. modify `head` elements) in the future without breaking changes.

**Comment by David-OConnor** [comment]
Working on a generic/trait-based approach where the view function could return either an El, or Vec of Els. This would be non-breaking, and allow either approach. (Notably, it wouldn't require you to wrap the top-level view fn in vec![] unless you wanted to return multiple as in your case)

**Comment by David-OConnor** [comment]
No luck with generic/trait approach. Latest commit has this new API where we accept `Vec<El>`, but the implementation still wraps in a div. Standby.

**Comment by MartinKavik** [comment]
@David-OConnor: What was the problem? Can I help you somehow?

**Comment by David-OConnor** [comment]
I ran out of time and had to go to work! I'll give it a shot this evening. No need for you to dive in yet. Need to refactor to mount the returned Els directly in the mount pt.

**Comment by David-OConnor** [comment]
Fixed in v0.3.1

**Comment by MartinKavik** [comment]
Well, I had a whole day for experimenting and here is the result (see mainly `view` in `examples/counter`):
https://github.com/David-OConnor/seed/compare/master...MartinKavik:88-view-return-vector

Some examples of `view` with that implementation:
**1.**
```rust
fn view(model: &Model) -> impl ElContainer<Msg> {
    div![]
}
```

**2.**
```rust
fn view(model: &Model) -> impl ElContainer<Msg> {
    vec! [
        div![],
        div![],
   ]
}
```

**3.**
```rust
fn view(model: &Model) -> El<Msg> {
    div![]
}
```

**4.**
```rust
fn view(model: &Model) -> Vec<El<Msg>> {
    vec! [
        div![],
        div![],
   ]
}
```
And because **4.** is the same as the current API, I think it's not a breaking change.

_Notes:_
I've changed `viewFn` to 
```rust
type ViewFn<Mdl, ElC> = fn(&Mdl) -> ElC;
```
because compiler ignores redundant type parameters and fails on `unused type parameter Ms`, e.g.:
```rust
type ViewFn<Ms, Mdl, ElC: ElContainer<Ms>> = fn(&Mdl) -> ElC;  // FAIL
```
and we can't use `impl` everywhere yet (see `initial limitations` in [RFC](https://github.com/rust-lang/rfcs/blob/master/text/1522-conservative-impl-trait.md)), e.g.:
```rust
type ViewFn<Ms, Mdl> = fn(&Mdl) -> impl ElContainer<Ms>;  // FAIL
```

---

_OT:_ @David-OConnor, I've setuped quickstart for better developer experience and production-ready output + deploy. I would like to know your opinion, thanks - https://github.com/MartinKavik/seed-quickstart-webpack.

**Comment by David-OConnor** [comment]
Haven't looked at the quickstart yet, but this appears to solve the problem we discussed, and would be non-breaking for both v <3.1 (view returns `El`), and 3.1 (View returns `Vec<El>`. If you make that as a PR, I'll merge it.

**Comment by David-OConnor** [comment]
Merged, and published as v0.3.2. Added your quickstart to the readme and guide. Great change - more flexible than either of the previous approaches.
